File: PHP/Program/Mahasiswa.php
<?php
/* Saya Indah Resti Fauzi NIM 2103507 mengerjakan soal Latihan 1 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin. */

require ('Civitas.php');
//Membuat class Mahasiswa
class Mahasiswa extends Civitas{
    //Private attributes dari class mahasiswa
    private $nim = '';
    private $fakultas = '';
    private $prodi = '';

    //Public methods dari class mahasiswa
    /*Constructors*/
    //Methods untuk set attributes dari class mahasiswa dengan parameter
    public function __construct($nik = '', $nama = '', $jenis_kelamin = '', $asal_universitas = '', $fakultas = '', $prodi = '', $nim = '', $email_edu = '')
    {
        //Set attributes pada class mahasiswa sesuai isi parameter
        $this->nik = $nik;
        $this->nama = $nama;
        $this->jenis_kelamin = $jenis_kelamin;
        $this->asal_universitas = $asal_universitas;
        $this->fakultas = $fakultas;
        $this->prodi = $prodi;
        $this->nim = $nim;
        $this->email_edu = $email_edu;
    }

    /*Getter dan Setter*/

    //Getter dan setter untuk nik mahasiswa
    public function setNik($nik){
        $this->nik = $nik;
    }
    public function getNik(){
        return $this->nik;
    }
    //Getter dan setter untuk nama mahasiswa
    public function setNama($nama){
        $this->nama = $nama;
    }
    public function getNama(){
        return $this->nama;
    }

    //Getter dan setter untuk jenis kelamin mahasiswa
    public function setJenis_Kelamin($jenis_kelamin){
        $this->jenis_kelamin = $jenis_kelamin;
    }
    public function getJenis_Kelamin(){
        return $this->jenis_kelamin;
    }

    //Getter dan setter untuk asal universitas mahasiswa
    public function setUniv($asal_universitas){
        $this->asal_universitas = $asal_universitas;
    }
    public function getUniv(){
        return $this->asal_universitas;
    }

    //Getter dan setter untuk fakultas mahasiswa
    public function setFakultas($fakultas){
        $this->fakultas = $fakultas;
    }
    public function getFakultas(){
        return $this->fakultas;
    }

    //Getter dan setter untuk prodi mahasiswa
    public function setProdi($prodi){
        $this->prodi = $prodi;
    }
    public function getProdi(){
        return $this->prodi;
    }

    //Getter dan setter untuk nim mahasiswa
    public function setNim($nim){
        $this->nim = $nim;
    }
    public function getNim(){
        return $this->nim;
    }

    //Getter dan setter untuk foto mahasiswa
    public function setEmail($email_edu){
        $this->email_edu= $email_edu;
    }
    public function getEmail(){
        return $this->email_edu;
    }
}

?>

File: PYTHON/Program/Mahasiswa.py
# Saya Indah Resti Fauzi NIM 2103507 mengerjakan soal Latihan 1 dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.
from Civitas import Civitas
# Membuat class Mahasiswa
class Mahasiswa (Civitas):
    
    # Attributes dari class mahasiswa
    __nim = ""
    __prodi = ""
    __fakultas = ""
    
    # Constructors
    # methods dari class mahasiswa
    def __init__(self, nik = "", nama = "", jenis_kelamin ="", asal_universitas = "", fakultas = "", prodi = "", nim = "", email_edu = ""):
        # Set attributes pada class mahasiswa
        self.__nama = nama
        self.__nim = nim
        self.__prodi = prodi
        self.__fakultas = fakultas
        self.__nik = nik
        self.__asal_universitas = asal_universitas
        self.__jenis_kelamin = jenis_kelamin
        self.__email_edu = email_edu


    # Getter dan Setter

    # Getter dan setter untuk nama mahasiswa
    def get_nama(self):
        return self.__nama
    def set_nama(self, nama):
        self.__nama = nama
        
    # Getter dan setter untuk nim mahasiswa
    def get_nim(self):
        return self.__nim
    def set_nim(self, nim):
        self.__nim = nim
        
    # Getter dan setter untuk prodi mahasiswa
    def get_prodi(self):
        return self.__prodi
    def set_prodi(self, prodi):
        self.__prodi = prodi
        
    # Getter dan setter untuk fakultas mahasiswa
    def get_fakultas(self):
        return self.__fakultas
    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas
        
    # Getter dan setter untuk nik mahasiswa
    def get_nik(self):
        return self.__nik
    def set_nik(self, nik):
        self.__nik = nik
        
    # Getter dan setter untuk nama mahasiswa
    def get_jenis(self):
        return self.__jenis_kelamin
    def set_jenis(self, jenis_kelamin):
        self.__jenis_kelamin = jenis_kelamin
        
    # Getter dan setter untuk nama mahasiswa
    def get_univ(self):
        return self.__asal_universitas
    def set_univ(self, asal_universitas):
        self.__asal_universitas = asal_universitas
        
    # Getter dan setter untuk nama mahasiswa
    def get_email(self):
        return self.__email_edu
    def set_email(self, email_edu):
        self.__email_edu = email_edu